### PR TITLE
TimeSeriesWidget support for multiple time series.

### PR DESCRIPTION
**Pull request by zbigg**
# Description

Shortcut: 
 * https://app.shortcut.com/cartoteam/story/336297/c4r-support-new-fields-model-for-timeseries-with-multiple-agg-splitbycategory-fidelity
 * https://app.shortcut.com/cartoteam/story/336298/c4r-implement-ui-for-timeseries-with-multiple-agg-splitbycategory

TimeSeriesWidget support for multiple time series. IN particular
 * support rendering of multiple series
 * support rendering for legend
 * support for new `splitByCategory` mode
 * support for filtering by category in `splitByCategory` mode
 * improved skeleton to better match sizing/placement of items with original (also in TableWidget)
 * new visuals in storybook
 

image

## Type of change

- Feature

# Acceptance

Please describe how to validate the feature or fix

1. go to X
2. test Y
3. assert Z

If feature deals with theme / UI or internal elements used also in CARTO 3, please also add a note on how to do acceptance on that part.

# Basic checklist

- Good PR name
- Shortcut link
- Changelog entry
- Just one issue per PR
- GitHub labels
- Proper status & reviewers
- Tests
- Documentation

### CODE REVIEW

File: packages/react-ui/src/widgets/TimeSeriesWidgetUI/TimeSeriesWidgetUI.js [hunk 1]
```diff
@@ -169,6 +192,63 @@ function TimeSeriesWidgetUIContent({
...
+    const series = [];
+
+    const categories = [];
+    const colorMapping = {};
+
+    // TODO: decide if we use common model for multiple series and splitByCategory
+    // as `splitByCategory` is less optimal and with common model we have "join" and then "split" data
+    // why common model?
+    // some other commponents are relying on old layout of data
+
```
Comment (zbigg) [inline packages/react-ui/src/widgets/TimeSeriesWidgetUI/TimeSeriesWidgetUI.js:205]: this is still open, probably i'll remove this code, comments welcome

File: packages/react-ui/src/widgets/TimeSeriesWidgetUI/TimeSeriesWidgetUI.js [hunk 2]
```diff
@@ -153,97 +184,76 @@ function TimeSeriesWidgetUIContent({
...
   height,
   showControls,
-  animation
+  animation,
+  palette,
+  selectedCategories,
+  onSelectedCategoriesChange,
+  showLegend
 }) {
-  const [anchorSpeedEl, setAnchorSpeedEl] = useState(null);
```
Comment (zbigg) [inline packages/react-ui/src/widgets/TimeSeriesWidgetUI/TimeSeriesWidgetUI.js:158]: animation handling moved _as_ is without changes to `TimeSeriesControls.js`

**Review by Josmorsot — APPROVED** [review]
lgtm

File: packages/react-ui/src/widgets/TimeSeriesWidgetUI/components/TimeSeriesControls.js [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+          onClick={handleTogglePlay}
+        >
+          {isPlaying ? <PauseIcon /> : <PlayIcon />}
+        </IconButton>
+      </Box>
+    </>
+  );
+}
+
+function ClockIcon() {
```
Comment (mdeividas) [inline packages/react-ui/src/widgets/TimeSeriesWidgetUI/components/TimeSeriesControls.js:184]: maybe it's worth considering to move those icons to separate files?

File: packages/react-ui/src/widgets/TimeSeriesWidgetUI/hooks/useTimeSeriesInteractivity.js [hunk 4]
```diff
@@ -46,7 +46,11 @@ export default function useTimeSeriesInteractivity({ echartsInstance, data }) {
   // Echarts events
   useEffect(() => {
     function clickEvent(params) {
-      if (!timeWindow.length && params.target?.type !== 'ec-line') {
+      // params target is specified if we hit data-line or point, not the background and for this,
+      // we
```
Comment (mdeividas) [inline packages/react-ui/src/widgets/TimeSeriesWidgetUI/hooks/useTimeSeriesInteractivity.js:50]: look like not finished comment?

File: packages/react-widgets/src/hooks/useWidgetFilterValues.js [hunk 5]
```diff
@@ -8,13 +8,14 @@ import { useMemo } from 'react';
  * @param  {object} props
  * @param  {string} props.dataSource - ID of the source to get the filters from.
  * @param  {string} props.id - ID of the widget that applied the filter.
- * @param  {string} props.column - name of column of this widget.
+ * @param  {string=} props.column - name of column of this widget.
```
Comment (mdeividas) [inline packages/react-widgets/src/hooks/useWidgetFilterValues.js:11]: do. you need this `=`? what does it mean?

**Review by mdeividas — APPROVED** [review]
in general, it looks good 👍

File: packages/react-widgets/src/hooks/useWidgetFilterValues.js (hunk 5 above)
Comment (zbigg) [reply packages/react-widgets/src/hooks/useWidgetFilterValues.js:11]: it's means _optional_ in JSDoc >> https://stackoverflow.com/questions/5873927/how-to-indicate-param-is-optional-using-inline-jsdoc

File: lerna.json [hunk 6]
```diff
@@ -3,6 +3,5 @@
     "packages/*"
   ],
   "npmClient": "yarn",
-  "useWorkspaces": true,
```
Comment (zbigg) [inline lerna.json:6]: new lerna complaints about that

File: packages/react-ui/src/widgets/TimeSeriesWidgetUI/components/TimeSeriesLayout.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
+import { Box, styled } from '@mui/material';
+import { BREAKPOINTS } from '../../../theme/themeConstants';
+import React from 'react';
+
+const Root = styled(Box)(() => ({
+  containerType: 'inline-size',
+  display: 'flex',
+  flexDirection: 'column',
+  border: '1px solid brown'
```
Comment (vmilan) [inline packages/react-ui/src/widgets/TimeSeriesWidgetUI/components/TimeSeriesLayout.js:9]: Is this border color right?

File: packages/react-ui/src/widgets/utils/colorUtils.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+/**
```
Comment (vmilan) [inline packages/react-ui/src/widgets/utils/colorUtils.js:1]: There is another file with color utils, maybe we can merge both files to have only one ¿?

packages/react-ui/src/utils/palette.js

**Review by vmilan — APPROVED** [review]
LGTM, just consider the comment about color utils files

File: packages/react-ui/src/widgets/TimeSeriesWidgetUI/components/TimeSeriesLayout.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+const BoxHorz = styled(Box)(() => ({
+  display: 'flex',
+  flexDirection: 'row'
+}));
+
+const ControlsBox = styled(Box)(({ theme }) => ({
+  flexShrink: 0,
+  marginLeft: 0,
+  paddingLeft: theme.spacing(1),
+  [`@container (max-width: ${BREAKPOINTS.XS}px)`]: {
```
Comment (zbigg) [inline packages/react-ui/src/widgets/TimeSeriesWidgetUI/components/TimeSeriesLayout.js:26]: this fails in JSDOM with `Error: Could not parse CSS stylesheet`

**Review by vmilan — APPROVED** [review]
LGTM

File: lerna.json [hunk 10]
```diff
@@ -4,5 +4,5 @@
   ],
   "npmClient": "yarn",
   "useWorkspaces": true,
-  "version": "2.2.7"
+  "version": "2.2.8-alpha.timeseries.21.0"
```
Comment (zbigg) [inline lerna.json:7]: of course, to be removed from patch

**Pull request merged**